### ISSUE TITLE
The invocations to `git grep` fails if the total length of filepaths causes the command line be over 32767 

### ISSUE THREAD

**Issue by lumberyard-employee-dm** [awslabs/git-secrets#192]
So in the git-secrets script below there is an invocation to `git grep` where it supplies the list of source controlled files to the command
https://github.com/awslabs/git-secrets/blob/80230afa8c8bdeac766a0fece36f95ffaa0be778/git-secrets#L116

If the aggregate list of the file paths supplied to the git grep command causes the comamnd line length to be larger than 32767 characters on Windows which is the [max command line length](https://devblogs.microsoft.com/oldnewthing/20031210-00/?p=41553)
The invocation fails with the following error
```
C:\Users\<user name>\git-secrets\git-secrets: line 116: /mingw64/libexec/git-core/git: Argument list too long
```
This can occur with a merge from a large code base where the number of files and paths to those files causes the file list that would be supplied to `git grep` to be large.
Having several hundred files that are part of the merge that has a relative path similar to filepaths as `Code/CoreLibrary/EditorViewport/ViewportFile.ext` will eventually cause the single the `git grep` command to be over the 32767 line length
 
Now using "echo" and "xargs" allows the git-grep command to be invoked multiple times without issue
`GREP_OPTIONS= LC_ALL=C echo "${files[@]}" | xargs git grep -nwHEI ${options} "${combined_patterns}"`

Now the caveat with using xargs, is that when any of invocations that it spawns off returns a return code between 1-125, xargs returns the return code of 123
This is explained in the xargs man page [here](https://man7.org/linux/man-pages/man1/xargs.1.html#EXIT_STATUS)

That then exposes another bug in the git-secrets `process_output()` function where if the return code isn't 0 or 1, it doesn't actually output the error message. It just invokes exit with the status code
https://github.com/awslabs/git-secrets/blob/80230afa8c8bdeac766a0fece36f95ffaa0be778/git-secrets#L131-L145

I believe the fix for that is to actually add an echo statement that outputs to stderr before invoking the exit command.
`echo ${output} >&2`

**Comment by jrn** [comment]
I'm guessing https://github.com/awslabs/git-secrets/pull/153 would help with this.

**Comment by lumberyard-employee-dm** [comment]
Yes, after making this PR, I found out that PR #153 already existed.
This PR only covers `git_grep` function, where #153 covers `regular_grep` as well.
